numpy/core/tests/test_einsum.py

```diff
@@ -744,6 +746,53 @@ def test_einsum_all_contig_non_contig_output(self):
         np.einsum('ij,jk->ik', x, x, out=out)
         assert_array_equal(out.base, correct_base)
 
+    @pytest.mark.parametrize("dtype",
+             np.typecodes["AllFloat"] + np.typecodes["AllInteger"])
+    def test_different_paths(self, dtype):
+        # Test originally added to cover broken float16 path: gh-20305
+        # Likely most are covered elsewhere, at least partially.
+        # Note that it is plausible that this test needs to be relaxed for
+        # floating point round-off errors.
+        dtype = np.dtype(dtype)
+        # Simple test, designed to excersize most specialized code paths,
+        # note the +0.1 for floats.
+        arr = (np.arange(7) + 0.1).astype(dtype)
+        scalar = np.array(2, dtype=dtype)
+
+        # contig -> scalar:
+        res = np.einsum('i->', arr)
+        assert_allclose(res, dtype.type(21.7))
+        # contig, contig -> contig:
+        res = np.einsum('i,i->i', arr, arr)
+        assert_array_equal(res, arr * arr)
+        # noncontig, noncontig -> contig:
+        res = np.einsum('i,i->i', arr.repeat(2)[::2], arr.repeat(2)[::2])
+        assert_array_equal(res, arr * arr)
+        # contig + contig -> scalar
+        assert np.einsum('i,i->', arr, arr) == (arr * arr).sum()
```

Comment:
This is failing with `AssertionError: assert 95.270004 == 95.27`

Reply:
Also with `AssertionError: assert 95.27 == 95.26999999999998`''

The problems are on arm64 and s390x.

Reply:
arg, yeah it is a bit annoying: To trigger the issue I have to use non-integers but these rounding issues are tricky with pairwise summations.

But: I just realized using `+0.5` should work, should give exact results but trigger the issue well enough.

Reply:
Thanks, hopefully fixed now (also stripped the comment that this might need to be fixed up for round-offs, should be safe now)